api/submit_result.go

```diff
@@ -32,12 +36,32 @@ func (s *resultSubmitter) Submit(executionID string, outputKey string, outputDat
 	if err != nil {
 		return err
 	}
-	if err := exec.Complete(outputKey, outputData); err != nil {
-		return err
+
+	// parse output data into a map.
+	var outputDataMap map[string]interface{}
+	switch d := outputData.(type) {
+	case map[string]interface{}:
+		outputDataMap = d
+	case string:
```

Comment:
Pass json as slice of []bytes as it's a standard way for passing json data. it's standard because every pkg handles json expected bytes. Moreover, convert []bytes to string generates small overhead

Reply:
done